### ISSUE TITLE
drush_sitealias_local_site_path ignores a present sites.php

### ISSUE THREAD

**Issue by jurgenhaas** [drush-ops/drush#339]
Original title: Batch processes and alias groups are not working together

If there is an alias group in e.g. group.aliases.drush.inc and one is running a drush command which requires batch processing, then things get messed up in a way that Drush executes sub-processes for the wrong domain/alias.

Example:

```
drush @group updatedb
```

This starts the update for the first alias in the group and that hangs for ever. When looking into the process list, you can find this:

```
22295 0:18 /usr/bin/php -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/share/php/drush/drush.php --php=/usr/bin/php --php-options= -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off @group updatedb
22338 0:00 sh -c /usr/bin/php  -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/share/php/drush/drush.php --php=/usr/bin/php --php-options=' -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off'  --backend=2 --uri=www.alias1.com --root=/var/www/live/htdocs --reserve-margin=26  updatedb 2>&1
22339 5:43 /usr/bin/php -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/share/php/drush/drush.php --php=/usr/bin/php --php-options= -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off --backend=2 --uri=www.alias1.com --root=/var/www/live/htdocs --reserve-margin=26 updatedb
25729 0:00 sh -c /usr/bin/php  -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/share/php/drush/drush.php --php=/usr/bin/php --php-options=' -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off'  --backend=2 --yes --root=/var/www/live/htdocs --uri=www.alias2.com  updatedb-batch-process 37 37 2>&1
25730 0:00 /usr/bin/php -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off /usr/share/php/drush/drush.php --php=/usr/bin/php --php-options= -d magic_quotes_gpc=Off -d magic_quotes_runtime=Off -d magic_quotes_sybase=Off --backend=2 --yes --root=/var/www/live/htdocs --uri=www.alias2.com updatedb-batch-process 37 37
```

Then, looking into the database from a second console, go to the database of alias2 and do

```
select * from batch;
```

and you'll see the table is empty. Looking into the database of alias1 instead you'll see that the batch with id 37 is in there.

So, for some reason, the third sub-process (ID 22339) is calling the fourth sub-process (ID 25729) with the wrong URI parameter.

This is tested against Drush version 6.2.0

**Comment by weitzman** [comment]
Looks like sites.php handing in this function is present (but broken). Closing this as its task is complete.

**Comment by weitzman** [comment]
Was fixed in e2d234a5d1314f0370c952cca4972d34f166d92c

**Comment by weitzman** [comment]
However, I'm seeing that Drush looks for entries in sites.php in the form localhost:888 instead of 8888:localhost, as required by the sites.php docs. Has anyone tried this with a port number?
